docs/rfc/0000-application-architecture.md

```diff
@@ -0,0 +1,503 @@
+
+* **Tier 1**:
+
+  > Fully supported platform, receives testing, deployment, and system
+  > configurations (may include packaging)
+
+  * Linux (`x86_64-unknown-linux-gnu`)
+  * macOS (`x86_64-apple-darwin`)
+
+* **Tier 2**:
+
+  > Basic support ("it compiles")
+
+  * Windows
+  * Other processor architectures for tier 1 platforms
+
+* **Tier 3**:
+
+  > Not supported, but potentially worth considering when assessing architecture
+  > decisions
+
+  * Mobile (iOS, Android)
+  * Wasm
+
+
+### System Services
+
+We will rely on declarative service / process management to be provided by the
+platform, specifically "socket activation".
+
+On macOS, this is provided on macOS via [launchd]. Linux distributions
+considered as tier 1 shall, however, by limited to those based on  [systemd] for
+the time being.
+
+We also assume the presence of an agent capable of producing Ed25519 signatures
+utilising the [SSH agent protocol][ssh-agent].
+
+On macOS, the distribution `ssh-agent` integrates with the system keychain.
+Other choices include the `ssh-agent` provided by the `openssh` suite, the
+`gpg-agent`, or (on Linux systems) `gnome-keyring`.
+
+Lastly, we consider [D-Bus][dbus] for publish-subscribe messaging. D-Bus is
+available on `systemd`-based systems per default, while it requires use of an
+external package manager on macOS.
+
+> Note: a D-Bus implementation is provided on `systemd`-based platforms, and can
+> be installed on macOS using `homebrew` or `MacPorts`. It may, however, turn
+> out to complicate builds due to varying link-time requirements. As we only
+> intend to use D-Bus for pub-sub messaging ("signals"), we may instead consider
+> to provide a purpose-built in-memory pub-sub service.
+
+
+### Signatures
+
+Each process which needs to produce signatures using the `radicle-link` _device
+key_, SHALL delegate this to an agent conforming to the [SSH agent
+protocol][ssh-agent] (section 4.5.).
+
+Key access restrictions thus depend on user configuration: the agent could be
+configured to prompt for a passphrase after a timeout elapses (or never),
+require presence of a hardware security token, or utilise platform
+authenticators such as TouchID&trade;.
+
+Key generation SHOULD, however, be performed using `radicle-link`-supplied
+tooling building on the [ed25519-zebra] library, which prevents certain weaker
+keys from being generated.
+
+Signature verification MUST be performed using the [ed25519-zebra] library, and
+thus MUST NOT be delegated to the agent.
+
+Note that (hypothetical) applications which require a large number of signatures
+to be generated (such as data migration tools) may find `ssh-agent` throughput
+insufficient, and require the raw key material to be supplied directly. This
+means that long-term secure key storage remains a concern of the `radicle-link`
+application suite.
+
+
+### Process Orchestration
+
+All daemon processes SHALL be started on-demand using the platform socket
```

Comment:
I think I'm left a bit confused as to what the "inner" architecture is. Is there some kind of "main" process that can spawn sub-processes, i.e. "daemon processes", worded here? Or when you say "daemon processes" do you mean multiple applications spinning up daemons?

Reply:
> Is there some kind of "main" process that can spawn sub-processes

`systemd`, `launchd`